### PR TITLE
Install airflow and providers from dist and verifies them 

### PR DESCRIPTION
**Pull request by potiuk**
This check is there to prevent problems similar to those reported
in #13027 and fixed in #13031.

Previously we always built airflow from wheels, only providers were
installed from sdist packages and tested. In this version both
airflow and providers are installed using the same package format
(sdist or wheel).

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: scripts/ci/images/ci_verify_ci_image.sh [hunk 1]
```diff
@@ -52,4 +52,6 @@ function pull_ci_image() {
 
 build_images::prepare_ci_build
 
+pull_ci_image
```
Comment (potiuk) [inline scripts/ci/images/ci_verify_ci_image.sh:55]: This is just to explicitly pull the image in separate pull command. I found out that I have not done that.

File: scripts/ci/images/ci_verify_prod_image.sh [hunk 2]
```diff
@@ -90,6 +90,8 @@ function pull_prod_image() {
 
 build_images::prepare_prod_build
 
+pull_prod_image
```
Comment (potiuk) [inline scripts/ci/images/ci_verify_prod_image.sh:93]: Same here.

File: Dockerfile [hunk 3]
```diff
@@ -255,7 +255,7 @@ RUN if [[ ${INSTALL_MYSQL_CLIENT} != "true" ]]; then \
         pip install --user ${ADDITIONAL_PYTHON_DEPS} --constraint "${AIRFLOW_CONSTRAINTS_LOCATION}"; \
     fi; \
     if [[ ${INSTALL_FROM_DOCKER_CONTEXT_FILES} == "true" ]]; then \
-        if ls /docker-context-files/*.whl 1> /dev/null 2>&1; then \
+        if ls /docker-context-files/*.{whl,tar.gz} 1> /dev/null 2>&1; then \
```
Comment (potiuk) [inline Dockerfile:258]: I also found out that I have sdist missing in ls check.

File: scripts/in_container/run_install_and_test_provider_packages.sh [hunk 4]
```diff
@@ -53,11 +53,20 @@ elif [[ ${INSTALL_AIRFLOW_VERSION} == "wheel"  ]]; then
     echo
     uninstall_airflow_and_providers
     install_airflow_from_wheel "[all]"
+    uninstall_providers
```
Comment (potiuk) [inline scripts/in_container/run_install_and_test_provider_packages.sh:56]: I also noticed that I did not uninstall providers here. It had no final effect before release (becuase it instlled rc1s first and then overwrote the installations with 1.0.0 versions built locally, but it is cleaner to remove the providers here

File: scripts/in_container/entrypoint_ci.sh [hunk 5]
```diff
@@ -103,7 +103,7 @@ if [[ -z ${INSTALL_AIRFLOW_VERSION=} ]]; then
     export PYTHONPATH=${AIRFLOW_SOURCES}
 elif [[ ${INSTALL_AIRFLOW_VERSION} == "none"  ]]; then
     echo
-    echo "Skip installing airflow - only install wheel packages that are present locally"
+    echo "Skip installing airflow - only install wheel/tar,gz packages that are present locally"
```
Comment (kaxil) [inline scripts/in_container/entrypoint_ci.sh]:
```suggestion
    echo "Skip installing airflow - only install wheel/tar.gz packages that are present locally"
```

**Review by kaxil — APPROVED** [review]

**Comment by potiuk** [comment]
We can always disable those later - it's as easy as changing matrix to only include wheel now.

**Pull request merged**